src/main/webapp/foodform.jsp

```diff
@@ -0,0 +1,34 @@
+<%-- The Java code in this JSP file runs on the server when the user navigates
+     to the homepage. This allows us to insert the Blobstore upload URL into the
+     form without building the HTML using print statements in a servlet. --%>
+<%@ page import="com.google.appengine.api.blobstore.BlobstoreService" %>
+<%@ page import="com.google.appengine.api.blobstore.BlobstoreServiceFactory" %>
+<% BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    String uploadUrl = blobstoreService.createUploadUrl("/my-form-handler"); %>
+
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="UTF-8">
+        <title>Log a food</title>
+    </head>
+    <body>
+        <p>Log what you ate and then click submit!</p>
+
+        <form method="POST" enctype="multipart/form-data" action="<%= uploadUrl %>">
+            <p>Upload an image:</p>
+            <input type="file" name="image">
+            <br/><br/>
+            <p>Date eaten:</p>
+            <textarea placeholder="mm/dd/yyyy" name="date" required></textarea>
```

Comment:
I think for all of these text inputs I would prefer `<input type="text"></input>` over `<textarea>`. Normally `<textarea>` is reserved for long form text entry. For example if you had a form where people could submit feedback or something like that.

Reply:
Also, I would make the `Date eaten`, `Food name`, etc. labels (see here: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) for greater accessibility.